cupy/logic/truth.py

```diff
@@ -57,3 +57,103 @@ def any(a, axis=None, out=None, keepdims=False):
 
     assert isinstance(a, cupy.ndarray)
     return a.any(axis=axis, out=out, keepdims=keepdims)
+
+
+def in1d(ar1, ar2, assume_unique=False, invert=False):
+    """
+    Test whether each element of a 1-D array is also present in a second array.
+
+    Returns a boolean array the same length as ``ar1`` that is True
+    where an element of ``ar1`` is in ``ar2`` and False otherwise.
+
+    Args:
+        ar1 (cupy.ndarray): Input array.
+        ar2  (cupy.ndarray):
+            The values against which to test each value of ``ar1``.
+        assume_unique (bool, optional):
+            If True, the input arrays are both assumed to be unique, which
+            can speed up the calculation. Default is False.
+        invert(bool, optional):
+            If True, the values in the returned array are inverted (that is,
+            False where an element of ``ar1`` is in ``ar2`` and True otherwise)
+            Default is False.
+
+    Returns:
+        cupy.ndarray, bool: The values ``ar1[in1d]`` are in ``ar2``.
+
+    """
+    # TODO(UmashankarTriforce): Improve efficiency of len check
+    # Ravel both arrays, behavior for the first array could be different
+    # check for shape
+    if not(ar1.shape[-1]):
+        return cupy.array([], dtype=bool)
+    ar1 = cupy.asarray(ar1).ravel()
```

Comment:
Please remove `cupy.asarray()`. That would cause unexpected data transfer which could cause device synchronization.

Reply:
Fixed